src/Microsoft.Data/DataFrameBuffer.cs

```diff
@@ -3,45 +3,39 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-using System.Text;
 
 namespace Microsoft.Data
 {
     /// <summary>
-    /// A basic store to hold values in a DataFrame column. Supports wrapping with an ArrowBuffer
+    /// A basic mutable store to hold values in a DataFrame column. Supports wrapping with an ArrowBuffer
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public class DataFrameBuffer<T>
+    internal class DataFrameBuffer<T> : ReadOnlyDataFrameBuffer<T>
         where T : struct
     {
-        // TODO: Change this to Memory<T>
-        public Memory<byte> Memory { get; private set; }
+        private Memory<byte> _memory;
 
-        private readonly int _size;
+        public override ReadOnlyMemory<byte> ReadOnlyMemory => _memory;
 
-        private int Capacity => Memory.Length / _size;
-
-        public int MaxCapacity => Int32.MaxValue / _size;
+        public Memory<byte> Memory
+        {
+            get => _memory;
+        }
 
         public Span<T> Span
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get => MemoryMarshal.Cast<byte, T>(Memory.Span);
         }
 
-        public int Length { get; internal set; }
+        public DataFrameBuffer(int numberOfValues = 8) : base(numberOfValues) { }
```

Comment:
Why have a default value?

Reply:
Hmm, I guess I greedily allocated some space upfront. I'll investigate if I have some logic that depends on this somewhere, but I'd rather do that in a separate patch and get this in. Sound ok? If you agree, I'll open an issue